### PR TITLE
Fix import with Flax but without PyTorch

### LINKED ISSUE #685: Can't import `diffusers` with Flax but without PyTorch
### Describe the bug

I discovered this when testing #659. It is currently possible to `import diffusers` without any of PyTorch and Flax; the dummy classes are loaded. But when Flax is installed and PyTorch isn't, then importing fails.

This is because of two reasons:

- `modeling_flax_utils.py` imports `load_state_dict`. This is so we can perform the weight conversion from a PyTorch checkout.
- Flax scheduler outputs are subclasses of `SchedulerOutput`, which declares the sample as a PyTorch tensor. I think we should create a `FlaxSchedulerOutput` instead.

### Reproduction

**Doesn't work**

```
pip uninstall torch
pip install flax

>>> import diffusers
```

**Works**
```
pip uninstall torch
pip uninstall flax

>>> import diffusers
```

### Logs

_No response_

### System Info

Diffusers @ `877bec8`

### PR DESCRIPTION
**Pull request by pcuenca**
Fixes #685.

### CODE REVIEW

**Review by pcuenca — COMMENTED** [review]
Here `SchedulerOutput` is conditionally defined to use either PyTorch tensors or jax arrays.

This is a little different to what we usually do: create an alternative class `FlaxSchedulerOutput`, load one or the other, and the corresponding dummy versions. But this approach felt easier and I think it would touch less files. Happy to change it to the other method if this is not a good idea.

File: src/diffusers/modeling_flax_utils.py [hunk 1]
```diff
@@ -391,6 +391,14 @@ def from_pretrained(
                 )
 
         if from_pt:
+            if is_torch_available():
+                from .modeling_utils import load_state_dict
+            else:
+                raise EnvironmentError(
+                    f"Can't load the model in PyTorch format because PyTorch is not installed. "
+                    f"Please, install PyTorch or use native Flax weights."
+                )
```
Comment (patil-suraj) [inline src/diffusers/modeling_flax_utils.py:400]: Good catch!

File: src/diffusers/schedulers/scheduling_utils.py [hunk 2]
```diff
@@ -14,26 +14,43 @@
...
+        Base class for the scheduler's step function output.
+
+        Args:
+            prev_sample (`jnp.ndarray` of shape `(batch_size, num_channels, height, width)` for images):
+                Computed sample (x_{t-1}) of previous timestep. `prev_sample` should be used as next model input in the
+                denoising loop.
+        """
 
-    prev_sample: torch.FloatTensor
+        prev_sample: jnp.ndarray
```
Comment (patil-suraj) [inline src/diffusers/schedulers/scheduling_utils.py:53]: Think we shouldn't define outputs classes like this. What if both PT and Flax is available ? This will override the definition no ? 

IMO let's add a new file called `scheduling_flax_utils` define flax related output classed and other flax stuff there.

Comment (pcuenca) [reply src/diffusers/schedulers/scheduling_utils.py:53]: You are right, I don't know what I was thinking. I'll do that.

**Review by patrickvonplaten — APPROVED** [review]
Very nice!

**Review by patil-suraj — COMMENTED** [review]
Thanks a lot! Looks good to me, my only comment is that we should `Flax` prefix to all those output classes for consistency and also to make it different from PT.

**Comment by pcuenca** [comment]
@patil-suraj I applied your suggestions but then found additional issues after installing on a fresh environment.

- If `transformers` is not installed import failed.
- The torch pipelines were imported when attempting to use Flax Stable Diffusion, so it didn't work.

**Review by patil-suraj — APPROVED** [review]
LGTM, thanks a lot!
